server/src/main/java/org/apache/kafka/server/purgatory/DelayedDeleteRecords.java

```diff
@@ -91,7 +91,8 @@ public boolean tryComplete() {
             }
         });
         //  check if every partition has satisfied at least one of case A or B
-        return deleteRecordsStatus.values().stream().noneMatch(DeleteRecordsPartitionStatus::acksPending) && forceComplete();
+        forceComplete();
```

Comment:
This change seems incorrect. We need to wait until there is no pending ack for all partitions before we can call forceComplete(), right?